python/cudf/cudf/_lib/cudf.pyx

```diff
@@ -173,15 +162,17 @@ cdef gdf_scalar* gdf_scalar_from_scalar(val, dtype=None) except? NULL:
     """
     cdef bool is_valid = True
 
+    if dtype is None:
+        dtype = val.dtype
```

Comment:
Is val always guaranteed to be a numpy scalar here?

Reply:
Done. Raising a `TypeError` now if it isn't